### DISCUSSION TITLE
Release 1.1

### DISCUSSION THREAD

**Discussion by ballsteve** [ballsteve/xrust#108] [Planning]
It's time to prepare the next release - 1.1. The namespace redesign is pretty massive, so it needs to get out there. Also, we should keep the release cadence as short as possible.

Let's bring everything into the dev branch, make sure all warnings are eliminated, and documentation is complete.

**Comment by Devasta** [comment]
Sounds good. I need to check if we have a fix for issue 98 now that the namespaces are updated, will try to get to it ASAP.

**Comment by ballsteve** [comment]
There's one test failing in the dev branch - conformance::xml::ibm_valid::ibmvalid_p66ibm66v01xml

Once that's fixed we can get the release out.

**Reply by Devasta** [reply]
Hi Steve,

I don't see this test failing?

**Reply by ballsteve** [reply]
Hmmm... there's a lot of newline characters in the test, so maybe it's a Mac/Unix thing. I'll investigate further.

**Reply by Devasta** [reply]
I only have access to a Windows laptop, but I can dig out an older box and put Linux on it for testing on that if needed.

**Reply by ballsteve** [reply]
I have a Mac laptop, so Unix O/S, and a Linux server so no problem with access to hardware & software.

**Reply by ballsteve** [reply]
I think I've found the problem: get_canonical translates \r\n into \n. This is being applied to the test document (P66/ibm66v01.xml) but not the output document (P66/out/ibm66v01.xml).

Why this is passing on a Windows box confuses me: it should fail everywhere.

Q: why is this translation being performed on output? Shouldn't it only be done on input?

**Reply by ballsteve** [reply]
Just checked the XML and Canonical XML specs.

XML (fifth edition), 2.11, says that CRLF should be translated to LF on input.

Canonical XML, 3.4, says that special characters, such as &#13; should be encoded as a character entity on output.

The change to the output is easy, and I'll do that tomorrow. I'd have to investigate the fix to the input parser.

We should add some additional tests for these, if there aren't already tests in the test suite.

**Reply by Devasta** [reply]
Grand, I'm on another business trip during the week but I can get to it at the weekend if needed.

**Comment by ballsteve** [comment]
Release 1.1 is now live!
